src/dtx/dtx_common.c

```diff
@@ -20,6 +20,12 @@
 struct dtx_batched_commit_args {
 	d_list_t		 dbca_link;
 	int			 dbca_refs;
+	uint32_t		 dbca_committing:1,
+				 dbca_aggregating:1,
+				 dbca_cleanup_stale:1;
+	struct sched_request	*dbca_cleanup_req;
+	struct sched_request	*dbca_commit_req;
+	struct sched_request	*dbca_agg_req;
```

Comment:
Minor: These flags indicating if ULT is running looks redundant to me.

They could be removed If you create a starting ULT wrapper function like:

static int start_ult(dbca, dbca->dbca_xxx_req) {
    if (dbca->dbca_xxx_req != NULL)
        return 0;
    dss_ult_create(&thread)
    if (error)
        return error;
    dbca->dbca_xxx_req = sched_req_get(thread);
    if (dbca->dbca_xxx_req == NULL) {
        ABT_thread_join(thread);
        return -DER_NOMEM;
    }
    return 0;
}

Reply:
I will refresh the patch for code cleanup.